dg/lib/ibmcloud/vlan_manager.py

```diff
@@ -0,0 +1,72 @@
+import json
+import logging
+
+from typing import Any, Optional
+
+from dg.lib.error import DataGateCLIException
+from dg.lib.ibmcloud import execute_ibmcloud_command
+from dg.utils.process import ProcessResult
+
+logger = logging.getLogger(__name__)
+
+
+class VLANManager:
+    """Determines available public and private VLANs for a zone in IBM
+    Cloud"""
+
+    def __init__(self, zone: str):
+        ibmcloud_oc_vlan_ls_command_args = ["oc", "vlan", "ls", "--json", "--zone", zone]
+        ibmcloud_oc_vlan_ls_command_result = execute_ibmcloud_command(
+            ibmcloud_oc_vlan_ls_command_args, capture_output=True
+        )
+
+        ibmcloud_oc_vlan_ls_command_result_json = json.loads(ibmcloud_oc_vlan_ls_command_result.stdout)
+        self._default_private_vlan = self._get_default_vlan_id(
+            ibmcloud_oc_vlan_ls_command_result,
+            ibmcloud_oc_vlan_ls_command_result_json,
+            "private",
+            zone,
+        )
+
+        self._default_public_vlan = self._get_default_vlan_id(
+            ibmcloud_oc_vlan_ls_command_result,
+            ibmcloud_oc_vlan_ls_command_result_json,
+            "public",
+            zone,
+        )
+
+    def get_default_private_vlan(self) -> str:
```

Comment:
as this is a getter, shouldn't you use `@property` and name the method itself `default_private_vlan`?
See e.g. https://stackoverflow.com/questions/6618002/using-property-versus-getters-and-setters.